test/expected/inner_fun_text.txt

```diff
@@ -18,14 +22,18 @@ local target="$2"
 mkdir -p "$target"
 if [[ -f "$1" ]]; then
 ln -s -f -t "$target" "$1"
-elif [[ -L "$1" ]]; then
-cp $1 $2
+elif [[ -L "$1" && ! -d "$1" ]]; then
+cp "$1" "$2"
 elif [[ -d "$1" ]]; then
-local children=$(find -H "$1" -maxdepth 1 -mindepth 1)
+SAVEIFS=$IFS
+IFS=$'
+'
```

Comment:
Technically newline is a legal (but highly unusual) file name character. Find supports -print0 to emit null characters which are the only illegal characters in file names (on Unix at least)

Reply:
I was thinking of making a similar change to this and yours but was going to use the following which handles spaces just fine.
```bash
find -H "$1" -maxdepth 1 -mindepth 1 | while read child; do
```

Do you have a sense of which approach might be better? Is there an additional change that should be made related to this PR?

Reply:
It really depends how pedantic you want to be about accepting valid but unusual file names. :-)

Your method covers the vast majority of cases, but it does miss some. It doesn’t properly handle trailing spaces or new lines in the file name. The null separator is something of a de facto Unix standard for handling this — several tools have built in support — but it can still be awkward to work with (see the weird input redirection in #443)

```
test$ find . | while read child; do echo "Child: [$child]"; ls -b "$child" ; done
Child: [.]
file1\ has\ whitespace                file4\ has\ trailing\ spaces\ \
file2\ \ has\ \ double\ \ whitespace  file5\ has\nnewline
\ \ file3\ has\ leading\ spaces       file\ has\ nonprintable\ \177
Child: [./file4 has trailing spaces]
ls: cannot access './file4 has trailing spaces': No such file or directory
Child: [./  file3 has leading spaces]
./\ \ file3\ has\ leading\ spaces
Child: [./file has nonprintable ]
./file\ has\ nonprintable\ \177
Child: [./file5 has]
ls: cannot access './file5 has': No such file or directory
Child: [newline]
ls: cannot access 'newline': No such file or directory
Child: [./file2  has  double  whitespace]
./file2\ \ has\ \ double\ \ whitespace
Child: [./file1 has whitespace]
./file1\ has\ whitespace
```